core/iwasm/common/wasm_application.c

```diff
@@ -404,8 +447,22 @@ wasm_application_execute_func(WASMModuleInstanceCommon *module_inst,
         goto fail;
     }
 
+#if WASM_ENABLE_REF_TYPES != 0
+    for (i = 0; i < type->param_count; i++) {
+        param_size_in_double_world += type_size_in_double_world(type->types[i]);
+    }
+    for (i = 0; i < type->result_count; i++) {
+        result_size_in_double_world +=
+            type_size_in_double_world(type->types[type->param_count + i]);
+    }
+    argc1 = param_size_in_double_world;
+    cell_num = (param_size_in_double_world >= result_size_in_double_world)
+                   ? param_size_in_double_world
+                   : result_size_in_double_world;
+#else
     argc1 = type->param_cell_num;
     cell_num = (argc1 > type->ret_cell_num) ? argc1 : type->ret_cell_num;
+#endif
```

Comment:
Should not re-calculate the param cell num here, we can use the type->param_cell_num when wasm_value_type_cell_num() is corretly modified.

Reply:
The size of an externref is different between inside of the call_WASM and outside of it. It is an uintptr_t outside and an uint32 inside. So existed cell_num variables are not correct here. Actually, this is the key point of the PR and made lots of things harder.

Reply:
OK, how about rename the function to type_cell_num_out_of_runtime, type_cell_num_of_native, or type_cell_num_of_xxx? And had better call wasm_value_type_cell_num for other types:
```C
static uint32
type_cell_num_out_of_runtime(uint8 type)
{
    if (type == VALUE_TYPE_EXTERNREF)
        return sizeof(uintptr_t) / sizeof(uint32);
    else
        return wasm_value_type_cell_num(type);
}